api/src/main/java/net/kyori/adventure/text/TextComponentImpl.java

```diff
@@ -69,19 +70,37 @@
 
   @Override
   public @NonNull TextComponent replace(final @NonNull Pattern pattern, final @NonNull UnaryOperator<Builder> replacement) {
+    return this.replace(pattern, replacement, index -> true);
```

Comment:
I feel like this and the two below should be default methods on the `TextComponent` interface, since they're just aliases, thoughts?

Reply:
Implemented